Backport from 1.6.3: apps which use the new dynamic get_init_creds_opt
structure don't get the same defaults as apps which use the old ones.
--- krb5-1.6.1/src/lib/krb5/krb/gic_opt.c	2007-02-05 18:45:17.000000000 -0500
+++ krb5-1.6.3/src/lib/krb5/krb/gic_opt.c	2007-10-01 22:42:02.000000000 -0400
@@ -1,9 +1,17 @@
 #include "k5-int.h"
+#include "int-proto.h"
+
+static void
+init_common(krb5_get_init_creds_opt *opt)
+{
+    opt->flags |= KRB5_GET_INIT_CREDS_OPT_CHG_PWD_PRMPT;
+}
 
 void KRB5_CALLCONV
 krb5_get_init_creds_opt_init(krb5_get_init_creds_opt *opt)
 {
-   opt->flags = KRB5_GET_INIT_CREDS_OPT_CHG_PWD_PRMPT;
+    opt->flags = 0;
+    init_common(opt);
 }
 
 void KRB5_CALLCONV
@@ -173,6 +181,7 @@
 	return ENOMEM;
 
     *opt = (krb5_get_init_creds_opt *) opte;
+    init_common(*opt);
     return 0;
 }
 
